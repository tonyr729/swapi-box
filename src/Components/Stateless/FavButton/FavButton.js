import React from 'react';
import './FavButton.css'
import PropTypes from 'prop-types'

const FavButton =({favorites, setDisplayedData}) => {

  return (
    <div className='favorite-div' onClick={() => setDisplayedData('favorites')}>
      <p className='favorite-text'>Favorites</p>
      <div className='favorite-count'>
        <p>
          {favorites}
        </p>
      </div>
    </div>
  )
}



FavButton.propTypes = {
  favorites: PropTypes.number,
  setDisplayedData: PropTypes.func
};

export default FavButton;