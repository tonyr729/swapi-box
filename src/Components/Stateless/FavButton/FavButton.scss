.favorite-text {
  color: #fff;
}

.favorite-count {
  color: #fff;
}