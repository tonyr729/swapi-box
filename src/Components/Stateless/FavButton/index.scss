p {
  margin: 0;
}

.favorite-div {
  position: absolute;
  right: 0;
  top: 0;
  &:hover { background-color: rgb(228, 148, 0); }
  &:active { background-color: rgb(255, 81, 1); }
}

.favorite-text {
  display: inline-block;
  color: #fff;
}

.favorite-count {
  display: inline-block;
  margin-left: 1rem;
  background-color: rgba(255, 255, 255, .5);
  color: #fff;
  padding: .3rem 1rem;
}