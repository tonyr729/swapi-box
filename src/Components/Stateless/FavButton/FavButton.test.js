import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import FavButton from './FavButton';

describe('FavButton', () => {

  describe('Component', () => {
    it('should match snapshot', () => {
      const favButton = renderer.create(<FavButton favorites={0}/>).toJSON();

      expect(favButton).toMatchSnapshot();
    })
  })
})