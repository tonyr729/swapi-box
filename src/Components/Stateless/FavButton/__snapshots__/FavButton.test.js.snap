// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavButton Component should match snapshot 1`] = `
<div
  onClick={[Function]}
>
  <p
    className="favorite-text"
  >
    Favorites
  </p>
  <div>
    <p
      className="favorite-count"
    >
      0
    </p>
  </div>
</div>
`;
